salt/_modules/caasp_hosts.py

```diff
@@ -271,12 +289,17 @@ def fqdn(name):
     def get_with_expr(expr):
         return __salt__['caasp_nodes.get_with_expr'](expr, grain='network.interfaces')
 
+    admin_nodes = admin_nodes or get_with_expr(ADMIN_EXPR)
+    master_nodes = master_nodes or get_with_expr(MASTER_EXPR)
+    worker_nodes = worker_nodes or get_with_expr(WORKER_EXPR)
+    other_nodes = other_nodes or get_with_expr(OTHER_EXPR)
+
```

Comment:
Using a common `add_names_for()` you could define here a new function:

```Python
    def _add_names_for_all(getter):
        for x in [admin_nodes, master_nodes, worker_nodes, other_nodes]:
            _add_names_for(hosts, x, infra_domain, getter)
```

so you could use a `_add_names_for_all(_get_primary_ip)` and a  `_add_names_for_all(_get_nodename)` later on...